lib/slashdeploy/service.rb

```diff
@@ -119,6 +119,13 @@ def lock_environment(user, environment, options = {})
       if lock
         return if lock.user == user # Already locked, nothing to do.
         fail EnvironmentLockedError, lock unless options[:force]
+        if environment['account']
+          direct_message \
```

Comment:
I think the best way to determine what Slack account to send to would be to add the following method to `app/models/lock.rb`:

```ruby
def slack_account
  environment.slack_account_for(user)
end
```

Then you can just use that here:

```ruby
direct_message lock.slack_account, ...
```

Reply:
What happens if `slack_account_for` gives me nothing? I added that check because there were tests failing for not having an account.

Reply:
If there's no account (e.g. `slack_account` returns nil), then `direct_message` will noop:

https://github.com/remind101/slashdeploy/blob/1e7d513cb0135e8a52ab21521781e36c78d5cf43/lib/slashdeploy/service.rb#L16-L18